Detectors/ITSMFT/common/reconstruction/include/ITSMFTReconstruction/RawPixelDecoder.h

```diff
@@ -156,6 +159,37 @@ int RawPixelDecoder<Mapping>::fillDecodedDigits(DigitContainer& digits, ROFConta
   return nFilled;
 }
 
+template <class Mapping>
+template <class DigitContainer, class ROFContainer>
+int RawPixelDecoder<Mapping>::fillDecodedDigitsHW(DigitContainer& digits, ROFContainer& rofs)
+{
+  if (mInteractionRecord.isDummy()) {
+    return 0; // nothing was decoded
+  }
+  mTimerFetchData.Start(false);
+  int ref = digits.size();
+  for (unsigned int iru = 0; iru < mRUDecodeVec.size(); iru++) {
+    uint16_t calcounter = mRUDecodeVec[iru].calCount;
+    uint16_t ninjection = mRUDecodeVec[iru].nInj;
+    uint16_t chargeinjected = mRUDecodeVec[iru].chargeInj;
+    uint16_t feeID = mMAP.RUSW2FEEId(mRUDecodeVec[iru].ruSWID, 0);
+    uint16_t half = (feeID >> 6) & 0x1;
+    uint16_t disk = (feeID >> 3) & 0x7;
+    uint16_t plane = (feeID >> 2) & 0x1;
+    uint16_t zone = feeID & 0x3;
```

Comment:
This will not work for ITS. If the ITS needs similar injection info, then it should be implemented in such a way that it works for both ITS and MFT. @iouribelikov could you somebody who is working on ITS calibration to clarify how they pass this info?

Reply:
I don't know if ITS needs them. But MFT does. There should be an easy way to provide specialisation to the general interface.

Reply:
This way of decoding the feeID is wrong fro ITS, if you insist on having derived digits with redundant info, then this should be implemented in a specialized `int RawPixelDecoder<ChipMappingMFT>::fillDecodedDigitsHW(DigitContainer& digits, ROFContainer& rofs)` method.
````